src/odemis/gui/comp/overlay/world.py

```diff
@@ -111,6 +113,200 @@ def on_left_up(self, evt):
+            else:
+                # Move to selected point
+                StagePointSelectOverlay.on_dbl_click(self, evt)
+        else:
+            WorldOverlay.on_dbl_click(self, evt)
+
+    def on_left_down(self, evt):
+        """
+        Handle mouse left click down: Create/Move feature if feature tool is toggled,
+        otherwise let the canvas handle the event (for proper dragging)
+        """
+        if self.active:
+            v_pos = evt.Position
+            feature = self._detect_point_inside_feature(v_pos)
+            if self._mode == MODE_EDIT_FEATURES:
+                if feature:
+                    # move/drag the selected feature
+                    self._selected_feature = feature
+                    DragMixin._on_left_down(self, evt)
+                else:
+                    # create new feature based on the physical position then disable the feature tool
+                    p_pos = self.cnvs.view_to_phys(v_pos, self.cnvs.get_half_buffer_size())
+                    self.cnvs.on_new_feature_pos(p_pos)
+                    self._selected_tool_va.value = TOOL_NONE
+            else:
+                if feature:
+                    self._current_feature_va.value = feature
+                self.cnvs.on_left_down(evt)
+        else:
+            WorldOverlay.on_left_down(self, evt)
+
+    def on_left_up(self, evt):
+        """
+        Handle mouse click left up: Move the selected feature to the designated point,
+        otherwise let the canvas handle the event when the overlay is active.
+        """
+        if self.active:
+            if self.left_dragging:
+                if self._selected_feature:
+                    v_pos = evt.Position
+                    p_pos = self.cnvs.view_to_phys(v_pos, self.cnvs.get_half_buffer_size())
+                    self._selected_feature.pos.value = tuple((p_pos[0], p_pos[1], self._selected_feature.pos.value[2]))
+                    self._selected_feature = None
+                    self._selected_tool_va.value = TOOL_NONE
+                    self.cnvs.update_drawing()
+                DragMixin._on_left_up(self, evt)
+            else:
+                self.cnvs.on_left_up(evt)
+        else:
+            WorldOverlay.on_left_up(self, evt)
+
+    def _detect_point_inside_feature(self, v_pos):
+        """
+        Detect if a given point is over a feature
+        :param v_pos: (int, int) Point in pixels
+        :return: (CryoFeature or None) Found feature, None if not found
+        """
+        def in_radius(c_x, c_y, r, x, y):
+            return math.hypot(c_x - x, c_y - y) <= r
+
+        offset = self.cnvs.get_half_buffer_size()  # to convert physical feature positions to pixels
+        for feature in self._features_va.value:
+            pos = feature.pos.value
+            fvsp = self.cnvs.phys_to_view(pos, offset)
+            if in_radius(fvsp[0], fvsp[1], FEATURE_DIAMETER, v_pos[0], v_pos[1]):
```

Comment:
It doesn't have to be a big block, @Mahmood-B  I made it as a function here because I'm not sure yet this is what is only needed to detect a feature. So in case there was a requirement to change/add more criteria, it'd be added in this small function.